### ISSUE TITLE
Data leakage

### ISSUE THREAD

**Issue by dylanjcastillo** [herniqeu/extract0#1]
Hey @herniqeu,

I made a comment on HN, but thought of sharing it here, in case you don't see it:

During SFT, it uses the full training dataset ([1]):
```
df = pd.read_csv('data/extraction_training_data.csv')
```

And during the evaluation, it uses the middle part of the same dataset ([2]):
```
df = pd.read_csv('data/extraction_training_data.csv')
df = df[100000:100000+NUM_TEST_SAMPLES]
```

Also, you split train/test/val by chunk and not by document ([3]). Then, the model "has seen" the documents that you're using to evaluate it (even if you're not evaluating it on the same chunks).

[1]: https://github.com/herniqeu/extract0/blob/0f8696a6fb1b620658ae6c22fd054cf422b4e9e9/src/evaluation/evaluate_model.py#L210

[2]: https://github.com/herniqeu/extract0/blob/0f8696a6fb1b620658ae6c22fd054cf422b4e9e9/src/training/supervised_finetuning.py#L85

[3]: https://github.com/herniqeu/extract0/blob/0f8696a6fb1b620658ae6c22fd054cf422b4e9e9/src/training/supervised_finetuning.py#L158

**Comment by herniqeu** [comment]
Hey @dylanjcastillo ,
Thanks for catching this. You're absolutely right about the data leakage in the current repository code.

Important clarification: The code differs from the paper's experimental setup only in this specific aspect. I modified it after publication to allow users to train on the full dataset if they want, but this wasn't how the paper's experiments were actually conducted.
Paper's actual experimental setup:

- 1,000 samples were held out as a test set before any training began
- SFT training used ~10K examples (not the full dataset)
- GRPO training used ~1K examples

All evaluation was conducted exclusively on the held-out test set that the model never saw during training

**Comment by dylanjcastillo** [comment]
Thank you. But what about splitting by chunks and not by documents? @herniqeu
